### PR TITLE
Context api dummy

### PR DESCRIPTION
**Pull request by satac2**
A minimal implementation of the context api with dummy methods, to be overridden by the SDK.

### CODE REVIEW

File: api/include/opentelemetry/context/context.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// This is a dummy class that is meant to be overridden,
+// the methods return default values.
+class Context
+{
+
+public:
+  // The Key class is used to obscure access from the
+  //  user to the context map. The identifier is used as a key
+  // to the context map.
+
```
Comment (adityashirodkar) [inline api/include/opentelemetry/context/context.h:22]: nit: remove newline

File: api/include/opentelemetry/context/context.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    char identifier_[50];
+  };
+
+  // Creates a key with the passed in name and returns it. 
+  Key CreateKey(nostd::string_view key_name){
+    return Key(key_name);
+  }
+
+  Context(){}
```
Comment (0x4b) [inline api/include/opentelemetry/context/context.h:50]: Google style is to use = default, but IDK if Open Telemetry has a preference
 
```suggestion
  Context() = default.
```

**Comment by satac2** [comment]
I am unsure about what to do with the templated methods, I feel as though I should leave them with the key_value_iterable guard, but I am not sure if that matters if the implementation does nothing anyway.

File: api/test/context/context_test.cc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+// Tests that the GetValue method does not crash
+TEST(ContextTest, ContextGetValueCrash)
+{
+  std::map<std::string, std::string> map_test = {{"test_key","123"}};
+  std::map<std::string, std::string> map_test_write = {{"foo_key","456"}};
+  Context context_test = Context(map_test);
+  Context::Key test_key = context_test.CreateKey("key_name");
+  context_test.GetValue(test_key);
```
Comment (adityashirodkar) [inline api/test/context/context_test.cc:46]: Expect 0

File: api/test/context/context_test.cc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  Context::Key test_key = context_test.CreateKey("key_name");
+  EXPECT_EQ(test_key.GetIdentifier(), "");
+}
+
+// Tests that the Context::Key GetName method does not crash
+TEST(ContextTest, ContextKeyGetNameCrash)
+{
+  Context context_test;
+  Context::Key test_key = context_test.CreateKey("key_name");
+  test_key.GetName();
```
Comment (adityashirodkar) [inline api/test/context/context_test.cc:62]: Expect empty

File: api/include/opentelemetry/context/context.h (hunk 1 above)
Comment (satac2) [reply api/include/opentelemetry/context/context.h]: Removed

File: api/test/context/context_test.cc (hunk 3 above)
Comment (satac2) [reply api/test/context/context_test.cc]: Added expect 0.

File: api/test/context/context_test.cc (hunk 4 above)
Comment (satac2) [reply api/test/context/context_test.cc]: Added expect "".

**Pull request closed without merging**